MyPerfectOnboardingApplication/Tests/MyPerfectOnboarding.Api.Tests/Controllers/ListControllerTests.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Web.Http;
+using MyPerfectOnboarding.Api.Controllers;
+using MyPerfectOnboarding.Api.Tests.Utils;
+using MyPerfectOnboarding.Contracts.Database;
+using MyPerfectOnboarding.Contracts.Models;
+using MyPerfectOnboarding.Contracts.Services.Location;
+using NSubstitute;
+using NUnit.Framework;
+
+namespace MyPerfectOnboarding.Api.Tests.Controllers
+{
+    [TestFixture]
+    internal class ListControllerTests
+    {
+        private ListController _listController;
+        private readonly ListItemEqualityComparer _listItemEqualityComparer = new ListItemEqualityComparer();
+
+        private readonly ListItem[] _items =
+        {
+            new ListItem
+            {
+                Id = new Guid("0B9E6EAF-83DC-4A99-9D57-A39FAF258CAC"),
+                Text = "aaaaa",
+                IsActive = false,
+                CreationTime = new DateTime(1589, 12, 3),
+                LastUpdateTime = new DateTime(1896, 4, 7)
+            },
+            new ListItem
+            {
+                Id = new Guid("11AC59B7-9517-4EDD-9DDD-EB418A7C1644"),
+                Text = "dfads",
+                IsActive = false,
+                CreationTime = new DateTime(4568, 6, 23),
+                LastUpdateTime = new DateTime(8569, 8, 24)
+            },
+        };
+
+        private IListRepository _repository;
+        private IUrlLocator _location;
+
+        [SetUp]
+        public void Init()
+        {
+            _repository = Substitute.For<IListRepository>();
+            _location = Substitute.For<IUrlLocator>();
+
+            _listController = new ListController(_repository, _location)
+            {
+                Request = new HttpRequestMessage(),
+                Configuration = new HttpConfiguration()
+            };
+        }
+
+        [Test]
+        public async Task Get_ReturnsListOfItems()
+        {
+            _repository.GetAllItemsAsync().Returns(Task.FromResult(_items));
+
+            var message = await _listController.ExecuteAction(controller => controller.GetAsync());
+            message.TryGetContentValue(out IEnumerable<ListItem> items);
+
+            Assert.That(message.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+            Assert.That(items, Is.EqualTo(_items).Using(_listItemEqualityComparer));
```

Comment:
I expected you to combine these two: https://github.com/HanaNavratilova/kentico-onboarding-cs/pull/3#discussion_r219792366 😆 

Anyway, it is true that you now repeat `.Using(_listItemEqualityComparer)` in this controller and You would surely be tempted to create another instance of the `ListItemEqualityComparer` were there another controller to be tested and your implementation is not `Lazy` (ask somebody why; I'm sure Juraj or Linda will know this time! 🤣).

However, I can wait with this detail until you finish the next task - since it now suffices.